### DISCUSSION TITLE
Contract negotiation with multiple assets

### DISCUSSION THREAD

**Discussion by RebeccaZirbs** [eclipse-edc/Samples#273] [Q&A]
Hello everyone, 

I would like to know if it is possible to extend the samples by creating multiple assets and combining them within a single contract negotiation? 
I would create several assets in the first step, execute step 2 and 3 normally, as it is explicitly mentioned there that it is realized in such a way that these steps contain all existing assets. But how can I adapt steps 4 and 5 within the contract negotiation so that I have negotiated all assets within the one contract and so that the agreement ID can be used for all assets?

Thanks in advance for your help!

**Comment by paullatzelsperger** [comment]
Contracts are always negotiated for exactly one asset. So no, it’s not possible to negotiate one contract for multiple assets. 

Note that contract definitions are fundamentally different from contract negotiations or contract agreements.
